clippy_tests/examples/range.stderr

```diff
@@ -1,11 +1,3 @@
-error: Range::step_by(0) produces an infinite iterator. Consider using `std::iter::repeat()` instead
```

Comment:
Actually, [`Iterator::step_by`](https://github.com/rust-lang/rust/pull/41439/files#diff-1398f322bc563592215b583e9b0ba936R290) has `assert!(step != 0);`, it will panic if the given step is `0`.